store/file/ods_q4.go

```diff
@@ -43,23 +43,21 @@ func CreateODSQ4(
 	go func() {
 		// doing this async shaves off ~27% of time for 128 ODS
 		// for bigger ODSes the discrepancy is even bigger
-		err := createQ4(pathQ4, eds)
-		if err != nil {
-			err = fmt.Errorf("сreating Q4 file: %w", err)
-		}
-
-		errCh <- err
+		errCh <- createQ4(pathQ4, eds)
 	}()
 
-	if err := CreateODS(pathODS, roots, eds); err != nil {
-		return fmt.Errorf("creating ODS file: %w", err)
-	}
+	err := CreateODS(pathODS, roots, eds)
+	q4Err := <-errCh
 
-	err := <-errCh
```

Comment:
Its waiting for the write in here. Where is the race exaclty?

Reply:
It was possible to return with error before reading from channel

Reply:
Update the description to clarify that

Reply:
Is it possible to test that? I mean add a test.

Reply:
Not sure how to reliably test it

Reply:
Unreliably the condition fails in tests added by https://github.com/celestiaorg/celestia-node/pull/3859

Reply:
Resolving 'cause not worth of additional testing. Thanks.